### PR TITLE
Bump net-ssh/scp for OpenSSL 3.0 support

### PR DESCRIPTION
**Pull request by agrare**
Fix `dh#set_pqg= is incompatible with OpenSSL 3.0`

https://github.com/net-ssh/net-ssh/pull/864

Depends on:
- [x] https://github.com/ManageIQ/linux_admin/pull/244
- [x] Release of linux_admin including :point_up: 

Fixes https://github.com/ManageIQ/manageiq-appliance_console/issues/245

### CODE REVIEW

File: manageiq-appliance_console.gemspec [hunk 1]
```diff
@@ -25,11 +25,13 @@ Gem::Specification.new do |spec|
...
   spec.add_runtime_dependency "bcrypt",                  "~> 3.1.10"
   spec.add_runtime_dependency "bcrypt_pbkdf",            ">= 1.0", "< 2.0"
+  spec.add_runtime_dependency "ed25519",                 ">= 1.2", "< 1.3"
   spec.add_runtime_dependency "highline",                "~> 2.1"
   spec.add_runtime_dependency "i18n",                    ">= 0.8"
   spec.add_runtime_dependency "linux_admin",             "~> 2.0"
   spec.add_runtime_dependency "manageiq-password",       "< 2"
-  spec.add_runtime_dependency "net-scp",                 "~> 1.2.1"
+  spec.add_runtime_dependency "net-scp",                 "~> 4.0"
+  spec.add_runtime_dependency "net-ssh",                 "~>7.2"
```
Comment (Fryguy) [inline manageiq-appliance_console.gemspec:34]:
```suggestion
  spec.add_runtime_dependency "net-ssh",                 "~> 7.2"
```

File: manageiq-appliance_console.gemspec [hunk 2]
```diff
@@ -25,11 +25,13 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "awesome_spawn",           "~> 1.6"
   spec.add_runtime_dependency "bcrypt",                  "~> 3.1.10"
   spec.add_runtime_dependency "bcrypt_pbkdf",            ">= 1.0", "< 2.0"
+  spec.add_runtime_dependency "ed25519",                 ">= 1.2", "< 1.3"
```
Comment (Fryguy) [inline manageiq-appliance_console.gemspec:28]: Out of curiosity why is this explicitly needed?

File: manageiq-appliance_console.gemspec (hunk 1 above)
Comment (Fryguy) [inline manageiq-appliance_console.gemspec:34]: @bdunne Please review - I'm curious if there are follow on effect of this large of a bump in both of these.

File: manageiq-appliance_console.gemspec (hunk 2 above)
Comment (agrare) [reply manageiq-appliance_console.gemspec:28]:
```
E, [2024-04-16T13:41:31.316966 #670326] ERROR -- net.ssh.authentication.agent[b18]: ignoring unimplemented key:unsupported key type `ssh-ed25519'
net-ssh requires the following gems for ed25519 support:
 * ed25519 (>= 1.2, < 2.0)
 * bcrypt_pbkdf (>= 1.0, < 2.0)
See https://github.com/net-ssh/net-ssh/issues/565 for more information
Gem::LoadError : "ed25519 is not part of the bundle. Add it to your Gemfile."
```

File: manageiq-appliance_console.gemspec (hunk 1 above)
Comment (agrare) [reply manageiq-appliance_console.gemspec]: Done

File: manageiq-appliance_console.gemspec (hunk 2 above)
Comment (Fryguy) [reply manageiq-appliance_console.gemspec:28]: ohhh interesting - wonder if we can drop the rbnacl gem then?

Comment (Fryguy) [reply manageiq-appliance_console.gemspec:28]: See https://github.com/ManageIQ/manageiq-appliance_console/commit/67653544afebe7bdbfedeed9c9161cac38592aeb - I think they were only added for this specific cipher.

Getting rid of those would be huge cause they require a C lib.

Comment (agrare) [reply manageiq-appliance_console.gemspec:28]: Added in 67653544afebe7bdbfedeed9c9161cac38592aeb

Looks like yes, that was the only reason rbnacl was added

We might be able to drop rbnacl-devel rpm also: 350c879fe1f080bf154bea3e0616d51b0c0d9ee6

Comment (agrare) [reply manageiq-appliance_console.gemspec:28]: I think we're okay actually, `jwt` just needs libsodium not `rbnacl`: https://github.com/ManageIQ/manageiq-rpm_build/pull/117#issuecomment-724800717

Comment (Fryguy) [reply manageiq-appliance_console.gemspec:28]: yeah it probably should have been there all along in both places.

Comment (agrare) [reply manageiq-appliance_console.gemspec:28]: Agreed, https://github.com/ManageIQ/manageiq-providers-google/pull/264

Comment (agrare) [reply manageiq-appliance_console.gemspec:28]: Removed here https://github.com/ManageIQ/manageiq-appliance_console/pull/248/files#diff-5cfba543196ff7509ccbd7b987182595a8baa8cc3e4a4e1040e20ccfd75d885eL36

**Review by Fryguy — APPROVED** [review]
LGTM but would like a second reviewer... @bdunne ?

File: manageiq-appliance_console.gemspec [hunk 3]
```diff
@@ -25,15 +25,16 @@ Gem::Specification.new do |spec|
...
```
Comment (bdunne) [inline manageiq-appliance_console.gemspec:28]: Is there a problem with 1.3?

Comment (agrare) [reply manageiq-appliance_console.gemspec:28]: Ah 1.3 was what they had in the description of the linked issue https://github.com/net-ssh/net-ssh/issues/565 but the actual documentation does have `< 2.0` I'll update to use that instead.

Comment (agrare) [reply manageiq-appliance_console.gemspec]: Done

**Comment by bdunne** [comment]
LinuxAdmin v3.0.0 is available https://rubygems.org/gems/linux_admin/versions/3.0.0

**Pull request merged**